arcaflow_plugin_pcp/pcp_schema.py

```diff
@@ -22,13 +22,79 @@ class InputParams:
     ] = None
 
 
+@dataclass
+class IntervalOutput:
+    interval: typing.Annotated[
+        int,
+        schema.id("@interval"),
+        schema.name("Interval ID"),
+        schema.description("The interval ID as reported by PCP"),
+    ]
+    timestamp: typing.Annotated[
+        str,
+        schema.id("@timestamp"),
+        schema.name("Interval timestamp"),
+        schema.description("The timestamp of the reported inverval"),
+    ]
+    commit: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("commit"),
+        schema.description("The commit for the interval"),
+    ] = None
+    disk: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("disk"),
+        schema.description("The disk structure for the interval"),
+    ] = None
+    kbin: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("kbin"),
+        schema.description("The KB in value for the interval"),
+    ] = None
+    kbout: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("kbout"),
+        schema.description("The KB out value for the interval"),
+    ] = None
+    kernel: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("kernel"),
+        schema.description("The kernel CPU structure for the interval"),
+    ] = None
+    mem: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("mem"),
+        schema.description("The memory structure for the interval"),
+    ] = None
+    memused: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("memused"),
+        schema.description("The memory used value for the interval"),
+    ] = None
+    pktin: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("pktin"),
+        schema.description("The packets in for the interval"),
+    ] = None
+    pktout: typing.Annotated[
+        typing.Optional[typing.Dict[str, typing.Any]],
+        schema.name("pktout"),
+        schema.description("The packets out for the interval"),
+    ] = None
+
+
+interval_output_schema = schema.ListType(
+    plugin.build_object_schema(IntervalOutput)
+)
+
+
 @dataclass
 class PerfOutput:
     pcp_output: typing.Annotated[
-        typing.Dict[str, typing.Any],
-        schema.name("PCP output dictionary"),
+        typing.List[IntervalOutput],
+        schema.name("PCP output list"),
         schema.description(
-            "Performance data from PCP provided in a " "JSON dictionary format"
+            "Performance data from PCP provided in a " "list format"
```

Comment:
While you're making this change, you might as well remove the " " that was probably caused by the linter moving two lines into one.

Reply:
fixed